tensorflow/python/ops/array_ops.py

```diff
@@ -2300,6 +2300,12 @@ def _constant_if_small(value, shape, dtype, name):
   return None
 
 
+def _eager_mark_zeros_tensor(tensor):
+  if context.executing_eagerly():
```

Comment:
Is there any harm to tagging non-EagerTensors? If not, we should just tag unconditionally. Then below we can check the property unconditionally too.